### PR TITLE
[Enhancement] Make the distribution balance threshold configurable

### PR DESCRIPTION
**Pull request by gengjun-git**
## Why I'm doing:
The balance of distribution will affect the balance of the disk, and a parameter is needed to control the impact.

## What I'm doing:
Introduce param `tablet_sched_distribution_balance_threshold_ratio`.
Doing distribution balance may break the disk size balance, but the maximum gap between disks cannot exceed `tablet_sched_distribution_balance_threshold_ratio * tablet_sched_balance_load_score_threshold`. If there are tablets in the cluster that are constantly balancing from A to B and B to A, reduce this value. If you want the tablet distribution to be more balanced, increase this value.

Fixes #SSU-1214

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by HangyuanLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -1477,6 +1475,15 @@ public class Config extends ConfigBase {
...
     public static long tablet_sched_consecutive_full_clone_delay_sec = 180; // 3min
 
+    /**
+     * Doing distribution balance may break the disk size balance,
+     * but the maximum gap between disks cannot exceed tablet_sched_distribution_balance_threshold_ratio * tablet_sched_balance_load_score_threshold
+     * If there are tablets in the cluster that are constantly balancing from A to B and B to A, reduce this value.
+     * If you want the tablet distribution to be more balanced, increase this value.
+     */
+    @ConfField(mutable = true)
+    public static double tablet_sched_distribution_balance_threshold_ratio = 0.5;
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1485]: how about `tablet_sched_num_based_balance_threshold_ratio`? balance can be number based or size based

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java]: done

**Review by nshangyiming — APPROVED** [review]

**Comment by gengjun-git** [comment]
@mergify rebase

**Review by HangyuanLiu — APPROVED** [review]

**Comment by gengjun-git** [comment]
@mergify backport branch-3.3

**Pull request merged**